test/integration/test_celery_user_rate_limit.py

```diff
@@ -0,0 +1,169 @@
+import datetime
+import tempfile
+from functools import lru_cache
+from typing import (
+    Dict,
+    Iterable,
+    List,
+)
+
+from celery.result import AsyncResult
+from sqlalchemy import text
+
+from galaxy.celery import galaxy_task
+from galaxy.model.database_utils import sqlalchemy_engine
+from galaxy.util import ExecutionTimer
+from galaxy_test.driver.driver_util import init_database
+from galaxy_test.driver.integration_util import IntegrationTestCase
+
+
+@galaxy_task(bind=True)
+def mock_user_id_task(self, task_user_id: int):
+    return task_user_id
+
+
+@lru_cache()
+def sqlite_url():
+    path = tempfile.NamedTemporaryFile().name
+    dburl = f"sqlite:///{path}"
+    init_database(dburl)
+    return dburl
+
+
+@lru_cache()
+def setup_users(dburl: str, num_users: int = 2):
+    """
+    Setup test users in galaxy_user table with user id's starting from 2 because
+    we assume there always exists a user with id = 1.
+    This is because the new celery_user_rate_limit table has
+    a user_id with a foreign key pointing to galaxy_user table.
+    """
+    expected_user_ids = [i for i in range(2, num_users + 1)]
+    with sqlalchemy_engine(dburl) as engine:
+        with engine.begin() as conn:
+            found_user_ids = conn.scalars(
+                text("select id from galaxy_user where id between 1 and :high"), {"high": num_users}
+            ).all()
+            if len(expected_user_ids) > len(found_user_ids):
```

Comment:
With the default value of `num_users` (which is what the tests are currently using), `len(expected_user_ids)` will be 1. Is that correct? So, when `num_users == 2`, and the database has only one user, this code won't create any additional users, and execute tasks for one user only. I would expect that the number of users tested would equal the value of the `num_users` parameter?

Reply:
It is a bug but it never caused a problem because at the time this code runs there apparently are no users in the galaxy_user table so len(found_user_ids) is always 0 and the condition len(expected_user_ids) > len(found_user_ids) is always true. Apparently the integration testing framework creates a brand new empty database everytime. The test assumes that there will always be a user id = 1, but apparently it is not present at the time this function runs and it somehow gets populated at some point after this code runs. I will change the code to say:
select id from galaxy_user where id between 2 and :high

Reply:
sometimes a new db is created, sometimes it is not - it depends on how the tests are run. I think, `run_tests.sh` will try to reuse the postgres database, whereas a direct execution via pytest will not, but I may be mistaken. 
The assumption about user.id=1 holds when the test uses the database that has been created by the testing framework. However, when a separate db is created for a test (like when we need to execute under sqlite regardless of the db env var passed to the test in the CI, the default user is not created. So, my suggestion would be to not make any assumptions about existing records in the db.